lib/rules/no-required-prop-with-default.js

```diff
@@ -0,0 +1,148 @@
+        return isKeyIdentifierEqual || isKeyLiteralEqual
+      })
+
+    let canAutoFix = false
+    const option = context.options[0]
+    if (option) {
+      canAutoFix = option.autofix
+    }
+
+    return utils.compositingVisitors(
+      utils.defineVueVisitor(context, {
+        onVueObjectEnter(node) {
+          utils.getComponentPropsFromOptions(node).map((prop) => {
+            if (
+              prop.type === 'object' &&
+              prop.propName &&
+              prop.value.type === 'ObjectExpression' &&
+              utils.findProperty(prop.value, 'default')
+            ) {
+              const requiredProperty = utils.findProperty(
+                prop.value,
+                'required'
+              )
+              if (!requiredProperty) return
+              const requiredNode = requiredProperty.value
+              if (
+                requiredNode &&
+                requiredNode.type === 'Literal' &&
+                !!requiredNode.value
+              ) {
+                context.report({
+                  node: prop.node,
+                  loc: prop.node.loc,
+                  data: {
+                    key: prop.propName
+                  },
+                  message: `Prop "{{ key }}" should be optional.`,
+                  fix: canAutoFix
+                    ? (fixer) => fixer.replaceText(requiredNode, 'false')
+                    : null
+                })
+              }
+            }
+          })
+        }
+      }),
+      utils.defineScriptSetupVisitor(context, {
+        onDefinePropsEnter(node, props) {
+          if (!utils.hasWithDefaults(node)) {
+            return
+          }
+          const withDefaultsProps = Object.keys(
+            utils.getWithDefaultsPropExpressions(node)
+          )
+          const requiredProps = props.flatMap((item) =>
+            item.type === 'type' && item.required ? [item] : []
+          )
+
+          for (const prop of requiredProps) {
+            if (withDefaultsProps.includes(prop.propName)) {
+              const firstToken = context
+                .getSourceCode()
+                .getFirstToken(prop.node)
+              // skip setter & getter case
+              if (firstToken.value === 'get' || firstToken.value === 'set') {
+                return
+              }
+              // skip computed
+              if (prop.node.computed) {
+                return
+              }
+              const keyToken = findKeyToken(
+                prop,
+                context.getSourceCode().getTokens(prop.node)
+              )
+              if (!keyToken) return
+              context.report({
+                node: prop.node,
+                loc: prop.node.loc,
+                data: {
+                  key: prop.propName
+                },
+                message: `Prop "{{ key }}" should be optional.`,
+                fix: canAutoFix
+                  ? (fixer) => fixer.insertTextAfter(keyToken, '?')
```

Comment:
I don't think we need to find the keyToken.

```suggestion
                  ? (fixer) => fixer.insertTextAfter(prop.key, '?')
```

Reply:
done